ts/features/idpay/payment/screens/IDPayPaymentAuthorizationScreen.tsx

```diff
@@ -59,39 +69,207 @@ const IDPayPaymentAuthorizationScreen = () => {
     machine.send("CONFIRM_AUTHORIZATION");
   };
 
-  // TODO Debug. Screen content will be added in another PR
-  const content = <Body>{JSON.stringify(transactionData, null, 4)}</Body>;
+  const renderContent = () => {
+    if (isLoading) {
+      return <TransactionDataSkeleton />;
+    }
+    if (transactionData !== undefined) {
+      return <ExistingTransactionDataContent data={transactionData} />;
+    }
+    // TODO:: error page, will be added in IOBP-6
+    return <></>;
+  };
 
   return (
     <BaseScreenComponent
       headerTitle="Autorizza operazione"
       contextualHelp={emptyContextualHelp}
+      goBack={false}
     >
-      <LoadingSpinnerOverlay isLoading={isLoading} loadingOpacity={100}>
-        <SafeAreaView style={IOStyles.flex}>
-          <View style={IOStyles.flex}>
-            <ContentWrapper>{content}</ContentWrapper>
-          </View>
-          <FooterWithButtons
-            type="TwoButtonsInlineHalf"
-            leftButton={{
-              title: "Annulla",
-              bordered: true,
-              onPress: handleCancel,
-              disabled: isUpserting
-            }}
-            rightButton={{
-              title: isUpserting ? "" : "Conferma",
-              onPress: handleConfirm,
-              isLoading: isUpserting,
-              disabled: isUpserting
-            }}
-          />
-        </SafeAreaView>
-      </LoadingSpinnerOverlay>
+      <SafeAreaView style={IOStyles.flex}>
+        <View style={IOStyles.flex}>
+          <ContentWrapper>
+            <H1>{I18n.t("idpay.payment.authorization.header")}</H1>
+            <VSpacer size={24} />
+            {renderContent()}
+          </ContentWrapper>
+        </View>
+        <FooterWithButtons
+          type="TwoButtonsInlineHalf"
+          leftButton={{
+            title: I18n.t("global.buttons.cancel"),
+            bordered: true,
+            onPress: handleCancel,
+            disabled: isUpserting
+          }}
+          rightButton={{
+            title: isUpserting ? "" : I18n.t("global.buttons.confirm"),
+            onPress: handleConfirm,
+            isLoading: isUpserting || isLoading,
+            disabled: isUpserting || isLoading
+          }}
+        />
+      </SafeAreaView>
     </BaseScreenComponent>
   );
 };
 
-export type { IDPayPaymentAuthorizationScreenRouteParams };
+const ExistingTransactionDataContent = ({
+  data
+}: {
+  data: NonNullable<AuthPaymentResponseDTO>;
+}) => {
+  const getValueOrDash = (value: string | undefined) =>
+    pipe(
+      value,
+      O.fromNullable,
+      O.getOrElse(() => "-")
+    );
+  const amountCents = formatNumberCurrency(data.amountCents / 100);
+  const reward = pipe(
+    data.reward,
+    O.fromNullable,
+    O.fold(
+      () => "-",
+      data => formatNumberCurrency(data / 100)
+    )
+  );
+  const businessName = getValueOrDash(data.businessName);
+  const initiativeName = getValueOrDash(data.initiativeName);
```

Comment:
```suggestion
  const businessName = data.businessName || "-";
  const initiativeName = data.initiativeName || "-";
```
Much simpler. What do you think?

Reply:
adressed in https://github.com/pagopa/io-app/pull/4680/commits/7191b70a5b16281c14883746efd231684f7bb32f